streaming/text/convert/c4.py

```diff
@@ -0,0 +1,109 @@
+import os
+from argparse import ArgumentParser, Namespace
+from typing import Any, Dict, Iterator
+
+import datasets
+from datasets.arrow_dataset import Dataset
+from torch.utils.data import DataLoader, IterableDataset, get_worker_info
+from tqdm import tqdm
+
+from ...base import MDSWriter
+
+
+def parse_args() -> Namespace:
+    """Parse commandline arguments.
+
+    Args:
+        Namespace: Commandline arguments.
+    """
+    args = ArgumentParser()
+    args.add_argument('--out', type=str, default='/datasets/mds/c4-zstd/')
+    args.add_argument('--compression', type=str, default='zstd:7')
+    args.add_argument('--hashes', type=str, default='sha1,xxh64')
+    args.add_argument('--limit', type=int, default=1 << 27)
+    args.add_argument('--batch_size', type=int, default=512)
+    args.add_argument('--progbar', type=int, default=1)
+    args.add_argument('--leave', type=int, default=0)
+    return args.parse_args()
+
+
+def get(split: str) -> IterableDataset:
+    """Collect the samples for this dataset split.
+
+    Args:
+        split (str): Split name.
+
+    Returns:
+        An IterableDataset.
+    """
+
+    class ShardedC4(IterableDataset):
+
+        def __init__(self):
+            self.dataset = datasets.load_dataset(  # pyright: ignore
+                path='c4', name='en', split=split, streaming=True)
+
+        def num_shards(self):
+            return len(self.dataset._ex_iterable.kwargs['filepaths'])  # pyright: ignore
+
+        def __iter__(self):
+            worker_info = get_worker_info()
+            if worker_info:
+                num_workers = worker_info.num_workers
+                worker_id = worker_info.id
+                shards = self.dataset._ex_iterable.kwargs['filepaths']  # pyright: ignore
+                assert len(shards) % num_workers == 0
```

Comment:
Recommendation: Can we please raise exception rather than `assert` as exception error code and error message is more descriptive and meaningful. Thanks!

Reply:
Done